### PR TITLE
Remove ATX counter

### LINKED ISSUE #1017: Optimize the block view selected for proving the active set size
# Overview / Motivation
Currently, when generating an ATX we choose all orphan blocks as the view that proves the active set size. We need to optimize the view in order to ease the traversal when validating the ATX.

# The Task
TODO: Clearly describe the issue requirements here...

# Implementation Notes
TODO: Add links to relevant resources, specs, related issues, etc...

# Contribution Guidelines
Important: Issue assignment to developers will be by the order of their application and proficiency level according to the tasks complexity. We will not assign tasks to developers who have'nt introduced themselves on our Gitter [dev channel](https://gitter.im/spacemesh-os/Lobby)

1. Introduce yourself on go-spacemesh [dev chat channel](https://gitter.im/spacemesh-os/Lobby) - ask our team any question you may have about this task
2. Fork branch `develop` to your own repo and work in your repo
3. You must document all methods, enums and types with [godoc comments](https://blog.golang.org/godoc-documenting-go-code)
4. You must write go unit tests for all types and methods when submitting a component, and integration tests if you submit a feature
5. When ready for code review, submit a PR from your repo back to branch `develop`
6. Attach relevant issue to PR

### PR DESCRIPTION
**Pull request by noamnelke**
See https://github.com/spacemeshos/go-spacemesh/pull/1217#issuecomment-517025687 for reasoning

Resolves https://github.com/spacemeshos/go-spacemesh/issues/1017

### CODE REVIEW

File: activation/activation.go [hunk 1]
```diff
@@ -312,33 +306,30 @@ func (b *Builder) PublishActivationTx(epoch types.EpochId) error {
...
+		if err != nil {
+			return fmt.Errorf("failed to get current view: %v", err)
+		}
 	}
-	atx := types.NewActivationTxWithChallenge(*b.challenge, b.coinbaseAccount, activeIds, view, b.nipst)
-	activeSetSize, err := b.db.CalcActiveSetFromView(atx) // TODO: remove this assertion to improve performance
 
-	if err != nil && !atx.TargetEpoch(b.layersPerEpoch).IsGenesis() {
-		b.log.Warning("empty active set size found! len(view): %d, view: %v", len(atx.View), atx.View)
+	activeSetSize, err := b.db.CalcActiveSetFromView(view, pubEpoch)
```
Comment (antonlerner) [inline activation/activation.go]: shouldn't we assert that active set size is > 0

File: activation/activation.go [hunk 2]
```diff
@@ -312,33 +306,30 @@ func (b *Builder) PublishActivationTx(epoch types.EpochId) error {
...
+	if err != nil && !targetEpoch.IsGenesis() {
```
Comment (antonlerner) [inline activation/activation.go]: Maybe error handling for this case should change? should we return the error?

Comment (noamnelke) [reply activation/activation.go]: _[relevant to both this comment and [the previous one](https://github.com/spacemeshos/go-spacemesh/pull/1232#discussion_r310530049)]_

Yes, good catch. Fixing this now.

File: activation/activation.go (hunk 1 above)
Comment (noamnelke) [reply activation/activation.go]: fixed

File: activation/activation.go (hunk 2 above)
Comment (noamnelke) [reply activation/activation.go]: fixed

File: activation/activation.go [hunk 3]
```diff
@@ -312,33 +306,36 @@ func (b *Builder) PublishActivationTx(epoch types.EpochId) error {
...
-	}
-	view, err := b.mesh.GetOrphanBlocksBefore(b.mesh.LatestLayer())
-	if err != nil {
-		return err
+	pubEpoch := b.challenge.PubLayerIdx.GetEpoch(b.layersPerEpoch) // can be 0 in first epoch
+	targetEpoch := pubEpoch + 1
+
+	viewLayer := pubEpoch.FirstLayer(b.layersPerEpoch)
+	var view []types.BlockID
+	if viewLayer > 0 {
```
Comment (zalmen) [inline activation/activation.go:314]: else...?

File: activation/activation.go [hunk 4]
```diff
@@ -312,33 +306,36 @@ func (b *Builder) PublishActivationTx(epoch types.EpochId) error {
...
+		var err error
+		view, err = b.mesh.GetOrphanBlocksBefore(viewLayer)
+		if err != nil {
+			return fmt.Errorf("failed to get current view: %v", err)
```
Comment (zalmen) [inline activation/activation.go]: also add `viewLayer` to the error

File: activation/activation.go (hunk 3 above)
Comment (noamnelke) [reply activation/activation.go:314]: if the `viewLayer` is 0 we should keep the view empty, as there are no blocks prior to this layer. First epoch ATXs will have an empty view and that's fine.

File: activation/activation.go (hunk 4 above)
Comment (noamnelke) [reply activation/activation.go]: done

**Review by zalmen — APPROVED** [review]

**Pull request merged**